concrete-core/src/backends/core/implementation/engines/destruction.rs

```diff
@@ -8,6 +8,7 @@ use crate::backends::core::implementation::entities::{
     PlaintextVector32, PlaintextVector64,
 };
 use crate::backends::core::private::math::tensor::AsMutTensor;
+use crate::prelude::{GgswCiphertext32, GgswCiphertext64, GgswCiphertextComplex64};
```

Comment:
@aPere3 tell me if I'm wrong but I think we want to have a `GgswCiphertextComplex32` as well, no? The idea is to differentiate between a complex ciphertext encrypting a 32 vs a 64 bits integer.

Reply:
They are called GgswCiphertextComplex64 because they use a concrete-core type called Complex64, which itself is an alias for a concrete-fftw type called c64. There is currently no Complex32 in concrete-core, but c32 exists in concrete-fftw, so I suppose it wouldn't be too much of a struggle to add them. Do we want to do it in this PR though?

Reply:
I'll come back to you with more detailed questions, but my first thought is that would require multiple, possibly breaking changes on the internal structure of GGSWs, so once again, should we do it in that PR?

Reply:
@orelcosseron I guess this implies adding a phantom type to the GgswCiphertext (the private one). @aPere3 I think it is ok to make the change here, I mean, the PR is already really big... what do you think?

Reply:
Oups, I did not realize that you were using `GgswCiphertext` to store data in the fourier domain. Yeah, before I would have created a separate `FourierGgswCiphertext` like you mention, but indeed, since the access pattern is the same, that would be a lot of duplication for not much.... I don't have strong opinion, because I don't know what we want to do with the internal codebase for now ....

For the problem of having both a `FourierGgswCiphertext32` and a `FourierGgswCiphertext64` entity available, maybe I am overlooking it, but I think you can just declare both of them like that:
```rust
pub struct GgswCiphertextComplex64(pub(crate) ImplGgswCiphertext<AlignedVec<Complex64>>);
....
pub struct GgswCiphertextComplex32(pub(crate) ImplGgswCiphertext<AlignedVec<Complex64>>);
```

We just have to take care about always making the `32` bit version operate with other `32` bit entities when we implement engine traits, but otherwise, I think it should work.

Reply:
Ah nope, this is definitely not something we would like to see :sweat_smile: ! So to answer you, the two structs would be __internally__ the same, but __not at all__ interchangeable, because we have two different types ! Let's take an example. We assume that we have:
```rust
impl GgswCiphertextConversionEngine<GgswCiphertext32, FourierGgswCiphertext32> for CoreEngine{...}
impl GgswCiphertextConversionEngine<GgswCiphertext64, FourierGgswCiphertext64> for CoreEngine{...}
```
Then, in client code, we could see:
```rust
let mut engine = CoreEngine::new().unwrap();
let ggsw: GgswCiphertext32 = ... 
let fourier_ggsw = engine.convert_ggsw(&ggsw).unwrap(); 
```
In this snippet, `fourier_ggsw` has type `FourierGgswCiphertext32` and can __not__ have the type `FourierGgswCiphertext64` because we never implemented `GgswCiphertextConversionEngine<GgswCiphertext32, FourierGgswCiphertext64>`. 

Actually we should take care about not implementing something like that, which what I meant by "always making the 32 bit version operate with other 32 bit entities" :sweat_smile: does that make sense ?